### ISSUE TITLE
Track trading revenue

### ISSUE THREAD

**Issue by bonomat** [get10101/10101#932]
We should track the revenue per trade/position, aka PnL.

**Comment by da-kami** [comment]
I think the best way to keep track of this state is extending the `Position` state to properly handle a `Closed` state.

This means that we update the position to `Closed` once it is closed and save the `realized` PnL (and potentially the closing price). I would save the PnL based on the DLC state, we could do that in a periodic task (note: the closing price would have to be saved upon `Closing` state and not when updating to `Closed`).

---

Note: I assume that tracking is only concerned about the *realized* PnL for reporting.
The unrealized PnL depends on the current price. We could periodically update the unrealized PnL in the database as well - is this something we would like to have @bonomat ?

**Comment by da-kami** [comment]
I assume that tracking is only concerned about the *realized* PnL for reporting.
The unrealized PnL depends on the current price. We could periodically update the unrealized PnL in the database as well - is this something we would like to have @bonomat ?

**Comment by bonomat** [comment]
Mhm... I would like to know the _unrealized_ PnL as well because this could give us an overview of what's happening before it's too late (aka realized PnL). 

I'm not sure what the best approach is for this.

**Comment by da-kami** [comment]
> Mhm... I would like to know the _unrealized_ PnL as well because this could give us an overview of what's happening before it's too late (aka realized PnL).
> 
> I'm not sure what the best approach is for this.

I already thought about it :)
I propose having two separate nullable fields in the `positions` table, one `realized_pnl` and one `unrealized_pnl`

The`realized_pnl` we update in the databse once we are in `Closed` state (before it is null in the database). The `unrealized_pnl` is set to null then.

The unrealized pnl we can periodically update in the database based on the current price. 
We have to be aware that the unrealized pnl is a snapshot of when the report was created. I think that's the best we can do.

Note: I would not depict the `unrealized_pnl` as part if the `Positions` model so far. The `unrealized_pnl` would only exist in the database (for reporting).
 
Theoretically we could also use the same `pnl` field for realized and unrealized and make it dependent on the state which one it is... - but I think such an "optimiztion" will cause more confusion than it helps :)

**Comment by da-kami** [comment]
Note: I already started adding the `realized_pnl` and the `Closed` state. I will see to add the unrealized part in a separate PR on top once this part is done :)
